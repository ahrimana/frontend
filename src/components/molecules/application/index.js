import React from 'react';
import { connect } from 'react-redux';
import Paper from 'material-ui/Paper';
import ApplicationIcon from 'material-ui/svg-icons/action/dashboard';
import actions from './actions/remove';
import styles from '../../atoms/icon/styles';


const mapStateToProps = (state) => ({
  theme: state.theme.theme,
});


export const Application = React.createClass({
  propTypes: {
    name: React.PropTypes.string.isRequired,
    serviceId: React.PropTypes.string.isRequired,
    confirm: React.PropTypes.func.isRequired,
  },

  handleClose(event) {
    event.preventDefault();
    const { serviceId, name } = this.props;
    const serviceUrl = `services/${serviceId}`;
    const url = `${serviceUrl}/applications/${name}`;
    this.props.confirm(url);
  },

  render() {
    return (
      <Paper style={styles.paper}>
        <div style={styles.close} onClick={this.handleClose}>x</div>
        <ApplicationIcon color={styles.icon.color} style={styles.icon} />
        <div>{this.props.name}</div>
      </Paper>
    );
  },
});


export default connect(mapStateToProps, actions)(Application);
