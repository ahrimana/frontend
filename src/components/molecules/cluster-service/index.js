import React from 'react';
import { connect } from 'react-redux';
import Paper from 'material-ui/Paper';
import ServiceIcon from 'material-ui/svg-icons/action/build';
import actions from './actions/remove';
import styles from '../../atoms/icon/styles';


const mapStateToProps = (state) => ({
  theme: state.theme.theme,
});


export const ClusterService = React.createClass({
  propTypes: {
    name: React.PropTypes.string.isRequired,
    clusterId: React.PropTypes.string.isRequired,
    serviceId: React.PropTypes.string.isRequired,
    confirm: React.PropTypes.func.isRequired,
  },

  handleClose(event) {
    event.preventDefault();
    const clusterUrl = `clusters/${this.props.clusterId}`;
    const url = `${clusterUrl}/services/${this.props.serviceId}`;
    this.props.confirm(url);
  },

  render() {
    return (
      <Paper style={styles.paper}>
        <div style={styles.close} onClick={this.handleClose}>x</div>
        <ServiceIcon color={styles.icon.color} style={styles.icon} />
        <div>{this.props.name}</div>
      </Paper>
    );
  },
});


export default connect(mapStateToProps, actions)(ClusterService);
