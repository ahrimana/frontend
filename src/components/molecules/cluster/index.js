import React from 'react';
import { connect } from 'react-redux';
import Paper from 'material-ui/Paper';
import ClusterIcon from 'material-ui/svg-icons/file/cloud-queue';
import actions from './actions/remove';
import styles from '../../atoms/icon/styles';


const mapStateToProps = (state) => ({
  theme: state.theme.theme,
});


export const Cluster = React.createClass({
  propTypes: {
    cluster: React.PropTypes.object.isRequired,
    confirm: React.PropTypes.func.isRequired,
  },

  handleClose(event) {
    event.preventDefault();
    this.props.confirm(this.props.cluster.id);
  },

  render() {
    return (
      <Paper style={styles.paper}>
        <div style={styles.close} onClick={this.handleClose}>x</div>
        <ClusterIcon color={styles.icon.color} style={styles.icon} />
        <div>{this.props.cluster.name}</div>
      </Paper>
    );
  },
});


export default connect(mapStateToProps, actions)(Cluster);
