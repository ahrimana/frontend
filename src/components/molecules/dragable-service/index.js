import React from 'react';
import { DragSource } from 'react-dnd';
import ItemTypes from './item-types';
import store from '../../../store';
import actions from './actions/add';


const serviceSource = {
  propTypes: {
    dispatch: React.PropTypes.func.isRequired,
  },

  beginDrag(props) {
    return {
      id: props.id,
    };
  },

  endDrag(props, monitor) {
    const item = monitor.getItem();
    const dropResult = monitor.getDropResult();
    const cluster = store.getState().clusterDetail.cluster;
    if (dropResult) {
      this.props.dispatch(actions.add(cluster.id, item.id));
    }
  },
};


const DragableService = React.createClass({
  propTypes: {
    name: React.PropTypes.string.isRequired,
    id: React.PropTypes.string.isRequired,
    connectDragSource: React.PropTypes.func.isRequired,
    isDragging: React.PropTypes.bool.isRequired,
  },

  render() {
    const { connectDragSource } = this.props;
    const { id } = this.props;
    return (
      connectDragSource(
        <div>{id}</div>
      )
    );
  },
});


// eslint-disable-next-line new-cap
export default DragSource(ItemTypes.SERVICE, serviceSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  isDragging: monitor.isDragging(),
}))(DragableService);
