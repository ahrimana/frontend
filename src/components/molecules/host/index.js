import React from 'react';
import { connect } from 'react-redux';
import Paper from 'material-ui/Paper';
import HostIcon from 'material-ui/svg-icons/hardware/memory';
import actions from './actions/remove';
import styles from '../../atoms/icon/styles';


const mapStateToProps = (state) => ({
  theme: state.theme.theme,
});


export const Host = React.createClass({
  propTypes: {
    name: React.PropTypes.string.isRequired,
    clusterId: React.PropTypes.string.isRequired,
    providerName: React.PropTypes.string.isRequired,
    confirm: React.PropTypes.func.isRequired,
  },

  handleClose(event) {
    event.preventDefault();
    const { clusterId, providerName, name } = this.props;
    const clusterUrl = `clusters/${clusterId}`;
    const providerUrl = `${clusterUrl}/providers/${providerName}`;
    const url = `${providerUrl}/hosts/${name}`;
    this.props.confirm(url);
  },

  render() {
    return (
      <Paper style={styles.paper}>
        <div style={styles.close} onClick={this.handleClose}>x</div>
        <HostIcon color={styles.icon.color} style={styles.icon} />
        <div>{this.props.name}</div>
      </Paper>
    );
  },
});


export default connect(mapStateToProps, actions)(Host);
