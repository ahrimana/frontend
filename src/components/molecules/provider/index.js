import React from 'react';
import { connect } from 'react-redux';
import Paper from 'material-ui/Paper';
import ProviderIcon from 'material-ui/svg-icons/device/storage';
import actions from './actions/remove';
import styles from '../../atoms/icon/styles';


const mapStateToProps = (state) => ({
  theme: state.theme.theme,
});


export const Provider = React.createClass({
  propTypes: {
    name: React.PropTypes.string.isRequired,
    clusterId: React.PropTypes.string.isRequired,
    confirm: React.PropTypes.func.isRequired,
  },

  handleClose(event) {
    event.preventDefault();
    const clusterId = this.props.clusterId;
    const name = this.props.name;
    const id = `clusters/${clusterId}/providers/${name}`;
    this.props.confirm(id);
  },

  render() {
    return (
      <Paper style={styles.paper}>
        <div style={styles.close} onClick={this.handleClose}>x</div>
        <ProviderIcon color={styles.icon.color} style={styles.icon} />
        <div>{this.props.name}</div>
      </Paper>
    );
  },
});


export default connect(mapStateToProps, actions)(Provider);
