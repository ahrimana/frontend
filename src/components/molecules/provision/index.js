import Log from '../../atoms/log';
import styles from './styles.js';
import radium from 'radium';
import React from 'react';

const Provision = React.createClass({
  propTypes: {
    provision: React.PropTypes.object,
  },

  getDefaultProps() {
    return {
      provision: {
        logs: [],
      },
    };
  },

  render() {
    const status = this.props.provision.status ? this.props.provision.status.toLowerCase() : '';
    return (
      <div>
        <h1 style={styles[status]}>Provision</h1>
        <Log>
          {
            this.props.provision.logs.map((log) => (
              <div key={log.timestamp} style={styles[log.status]}>
                [{log.host}] {log.task}{log.log ? `: ${log.log}` : ''}
              </div>
            ))
          }
        </Log>
      </div>
    );
  },
});


export default radium(Provision);
