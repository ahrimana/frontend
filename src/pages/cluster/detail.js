import React from 'react'
import PropTypes from 'prop-types'
import { observer } from 'mobx-react'

// Components
import Button from '@material-ui/core/Button'
import Paper from '@material-ui/core/Paper'
import TextField from '@material-ui/core/TextField'
import Tooltip from '@material-ui/core/Tooltip'

import store from 'store'
import Template from 'templates/default'
import styles from './styles'


@observer
class ClusterDetail extends React.Component {
  state = {
    edit: null,
  }

  componentWillMount() {
    store.title.title = 'Cluster Detail'
    store.cluster.fetch(this.props.match.params.id)
  }

  handleClick = (item) => () => {
    this.setState({
      edit: item,
      [item]: store.cluster.detail[item],
    })
  }

  handleCancel = (item) => () => {
    store.cluster.detail[item] = this.state[item]
    this.setState({
      edit: null,
      [item]: null,
    })
  }

  handleName = (event) => {
    store.cluster.detail.name = event.target.value
  }

  handleSubmit = async (event) => {
    event.preventDefault()
    const result = await store.cluster.edit(
      store.cluster.detail,
      { name: store.cluster.detail.name },
    )
    if (result.status >= 400) {
      store.error.message = result.error
      store.error.open = true
    }
    this.setState({
      edit: null,
    })
  }

  render() {
    const name = this.state.edit === 'name'
      ? (
        <form onSubmit={this.handleSubmit}>
          <TextField
            value={store.cluster.detail.name}
            onChange={this.handleName}
          />
          <Button
            variant="outlined"
            style={styles.button}
            color="primary"
            type="submit"
          >
            Save
          </Button>
          <Button
            variant="outlined"
            style={styles.button}
            color="secondary"
            onClick={this.handleCancel('name')}
          >
            Cancel
          </Button>
        </form>
      )
      : (
        <div
          onClick={this.handleClick('name')}
          role="button"
          onKeyDown={this.handleClick('name')}
          tabIndex={0}
        >
          <Tooltip placement="right" title="Click to edit">
            <h1 style={styles.name}>{store.cluster.detail.name}</h1>
          </Tooltip>
        </div>
      )
    return (
      <Template>
        <div style={styles.root}>
          <Paper style={styles.paper}>
            {name}
            <div style={styles.id}>
              id: {store.cluster.detail.id}
            </div>
          </Paper>
        </div>
      </Template>
    )
  }
}


ClusterDetail.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
}


export default ClusterDetail
