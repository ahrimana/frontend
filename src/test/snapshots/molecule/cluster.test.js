/* eslint no-undef: 0 */
/* eslint no-unused-expressions: 0 */
/* eslint no-sequences: 0 */
/* eslint quotes: 0 */

import React from 'react';
import { Cluster } from '../../../components/molecules/cluster';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("Cluster Molecule (snapshot)", () => {
  const setup = () => {
    const props = {
      cluster: {
        name: "wordpress",
      },
      confirm: jest.fn(),
    };
    return shallow(<Cluster {...props} />);
  };

  it("Cluster molecule render paper", () => {
    const component = setup();
    expect(toJson(component)).toMatchSnapshot();
  });
});
