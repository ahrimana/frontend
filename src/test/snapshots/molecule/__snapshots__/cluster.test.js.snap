exports[`Cluster Molecule (snapshot) Cluster molecule render paper 1`] = `
<Paper
  circle={false}
  rounded={true}
  style={
    Object {
      "borderRadius": 10,
      "display": "inline-block",
      "height": 120,
      "margin": 3,
      "textAlign": "center",
      "width": 120,
    }
  }
  transitionEnabled={true}
  zDepth={1}>
  <div
    onClick={[Function]}
    style={
      Object {
        "marginLeft": 100,
      }
    }>
    x
  </div>
  <FileCloudQueue
    color="#ccc"
    style={
      Object {
        "color": "#ccc",
        "height": "60%",
        "width": "100%",
      }
    } />
  <div>
    wordpress
  </div>
</Paper>
`;
